src/mgmt/services/manager/project/api/nodes.py

```diff
@@ -9,31 +9,196 @@
+                            droplet['spec']['ip'] = droplet_ip_list
+                            droplet['spec']['mac'] = droplet_mac_list
+                    if current_length_of_ip_list == len(ip_for_new_droplet):
+                        logger.info('Breaking the while loop as there is no new ip added')
+                        break
+                    else:
+                        current_length_of_ip_list = len(ip_for_new_droplet)
+                macs_for_new_droplet = [get_mac_from_ip(ip) for ip in ip_for_new_droplet]
+
+                for droplet_name in modified_droplets:
+                    update_droplet(modified_droplets[droplet_name])
+
+                create_droplet(name='droplet-tenant-' + post_data['name'].replace('_', "-").lower(), 
+                               ip=ip_for_new_droplet, 
+                               mac=macs_for_new_droplet,
+                               itf=post_data['inf_tenant'], 
+                               network='tenant', 
+                               zgc_id=post_data['zgc_id'])
+
+                create_droplet(name='droplet-zgc-' + post_data['name'].replace('_', "-").lower(), 
+                               ip=[zgc_ip], 
+                               mac=[zgc_mac],
+                               itf=post_data['inf_zgc'], 
+                               network='zgc', 
+                               zgc_id=post_data['zgc_id'])
+
+                logger.info('Finished updating and adding droplets.')
+            else:
+                logger.error('Not enough ip to assign for the new droplet.')
+        else:
+            logger.info("First nodes in the ZGC cluster, it gets all the IPs")
+            zgc = Zgc.query.filter_by(zgc_id=post_data['zgc_id']).first()
+            if zgc is not None:
+                gws = getGWsFromIpRange(zgc.ip_start, zgc.ip_end)
+                create_droplet(name='droplet-' +'tenant-' + post_data['name'].replace('_', "-").lower(), 
+                               ip=[gw['ip'] for gw in gws], 
+                               mac=[gw['mac'] for gw in gws],
+                               itf=post_data['inf_tenant'], 
+                               network='tenant', 
+                               zgc_id=post_data['zgc_id'])
+                create_droplet(name='droplet-' +'zgc-' + post_data['name'].replace('_', "-").lower(), 
+                               ip=[zgc_ip], 
+                               mac=[zgc_mac],
+                               itf=post_data['inf_zgc'], 
+                               network='zgc', 
+                               zgc_id=post_data['zgc_id'])
+            else:
+                logger.error("There's no zgc with zgc_id: {} in the database!".format(post_data['zgc_id']))
+                return jsonify({'error':'No such zgc'})
+        # commit change to data at last
         db.session.add(Node(**post_data))
         db.session.commit()
-        response_object = post_data
```

Comment:
don't make this change, see comment above

Reply:
fixed